File: src/3379.transformed-array.3651/solution.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int[] result = new int[nums.length];
        int L = nums.length;
        for(int i=0;i<L;i++){
            int index = (i+nums[i])%L;
            if(index<0){
                index += L;
            }
            result[i] = nums[index];
        }
        return result;
    }
}

File: src/3379.transformed-array.3651/solution.ts
function constructTransformedArray (nums: number[]): number[] {
    const result = new Array<number>(nums.length);
    const L = nums.length;
    for (let i = 0; i < L; i++) {
        result[i] = nums[((i + nums[i]) % L + L) % L];
    }
    return result;
}
